experimental/packages/otlp-exporter-base/src/types.ts

```diff
@@ -45,7 +45,7 @@ export interface ExportServiceError {
  * Collector Exporter base config
  */
 export interface OTLPExporterConfigBase {
-  headers?: Partial<Record<string, unknown>>;
```

Comment:
I don't have any context on why `Partial` was used here before, but is it a breaking change to remove it? I'm actually not sure

Reply:
Oh I see this was requested above https://github.com/open-telemetry/opentelemetry-js/pull/3662#discussion_r1162465074

Reply:
What would you feel is the best path for this? Should I 

1. indicate in change log that it's a breaking change
2. leave it as not a breaking change
3. put Partial back in again if there is a backward compatibility risk

?

Reply:
I think it's OK, I tried it out in [TS Playground](https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgMIHsQ2AcwPIA2AJsgN4BQyVyAFhHEdAM4D8AXMgApxRjBwEAPACUICdFCKCmYKKBwAaZAFcQAaxDoA7iAB8ugNzkAvuVCRYiFBiy4AchC1lK1Og2btko8ZOmz5Sqoa2nrIAD7IABSRAJTIALy6XmISUjJyIIrI6fK6MUam5DCqCHyYyJAykeK2OBw22PjEcRTUyAD07cgAyhAQALZMFejIAEYocExMuCBwowQQSprI0FASTC5UNTLIII4NuPWYjQ5O8cg1jQVAA) and it looks like the old config is assignable:

```ts
interface ConfigOld {
    headers?: Partial<Record<string, unknown>>;
}
interface ConfigNew {
    headers?: Record<string, unknown> | (() => Record<string, string>);
}

function test(config: ConfigOld) {
    // Seems to be assignable, no errors
    const newConfig: ConfigNew = config;
}
```